build.sh

```diff
@@ -0,0 +1,114 @@
+#!/usr/bin/env bash
+
+# build.sh - Build Jellyfin binary packages
+# Part of the Jellyfin Project
+
+set -o errexit
+set -o pipefail
+
+usage() {
+    echo -e "build.sh - Build Jellyfin binary packages"
+    echo -e "Usage:"
+    echo -e "  $0 -t/--type <BUILD_TYPE> -p/--platform <PLATFORM> [-k/--keep-artifacts] [-l/--list-platforms]"
+    echo -e "Notes:"
+    echo -e "  * BUILD_TYPE can be one of: [native, docker] and must be specified"
+    echo -e "    * native: Build using the build script in the host OS"
+    echo -e "    * docker: Build using the build script in a standardized Docker container"
+    echo -e "  * PLATFORM can be any platform shown by -l/--list-platforms and must be specified"
+    echo -e "  * If -k/--keep-artifacts is specified, transient artifacts (e.g. Docker containers) will be"
+    echo -e "    retained after the build is finished; the source directory will still be cleaned"
+    echo -e "  * If -l/--list-platforms is specified, all other arguments are ignored; the script will print"
+    echo -e "    the list of supported platforms and exit"
+}
+
+list_platforms() {
+    declare -a platforms
+    platforms=(
+        $( find deployment -maxdepth 1 -mindepth 1 -name "build.*" | awk -F'.' '{ $1=""; printf $2; if ($3 != ""){ printf "." $3; }; if ($4 != ""){ printf "." $4; };  print ""; }' | sort )
+    )
+    echo -e "Valid platforms:"
+    echo
+    for platform in ${platforms[@]}; do
+        echo -e "* ${platform} : $( grep '^#=' deployment/build.${platform} | sed 's/^#= //' )"
+    done
+}
+
+do_build_native() {
+    if [[ $( dpkg --print-architecture | head -1 ) != "${PLATFORM##*.}" ]]; then
+        echo "Cross-building is not supported for native builds, use 'docker' builds on amd64 for cross-building."
+        exit 1
+    fi
+    export IS_DOCKER=NO
+    deployment/build.${PLATFORM}
+}
+
+do_build_docker() {
+    if ! dpkg --print-architecture | grep -q 'amd64'; then
+        echo "Docker-based builds only support amd64-based cross-building; use a 'native' build instead."
```

Comment:
This one is checking that Docker builds are done in amd64. They're not supported for arm any more to avoid Dockerfile sprawl. For arm, native-only going forward.

Reply:
So, let's see what you have in store for building for ARM.

Reply:
I'm not sure what you mean?

On amd64, Docker and Native are supported.
On armX, Native only is supported.

Reply:
For posterity:

* If your host system is amd64, ARM builds can be done in two ways: in docker, and native. The latter requires you to already have set up the full crossbuild environment.
* If your host system is arm (hf or 64), ARM builds can only be done native, and do not require any crossbuild infra.

Complexity in flags, vs. complexity in matinaing a bunch of Dockerfiles